libr/include/r_esil.h

```diff
@@ -148,6 +148,29 @@ typedef struct r_esil_callbacks_t {
 	bool (*reg_write)(ESIL *esil, const char *name, ut64 val);
 } REsilCallbacks;
 
+typedef bool (*REsilMemSwitch)(void *mem, ut32 idx);
+typedef bool (*REsilMemRead)(void *mem, ut64 addr, ut8 *buf, int len);
+typedef bool (*REsilMemWrite)(void *mem, ut64 addr, const ut8 *buf, int len);
+
+typedef struct r_esil_memory_interface_t {
+	void *mem;
```

Comment:
userdata?

Reply:
yes, within r2 this would be RIO *

Reply:
Better user userdata instead of mem for cosistency

Reply:
will change that later